### ISSUE TITLE
Linux execution errors

### ISSUE THREAD

**Issue by BoatsAreRockable** [BoatsAreRockable/Boats-Animator#171]
Humblehacker on Bricks in Motion [has reported](http://www.bricksinmotion.com/forums/post/360156/#p360156) execution errors with the Linux build of BA in Ubuntu 15.10:

~~~~
[2926:0519/141150:ERROR:icu_util.cc(152)] Couldn't mmap /home/user/Boats-Animator-0.6.5-linux64/icudtl.dat
[2926:0519/141150:FATAL:content_main_runner.cc(749)] Check failed: base::i18n::InitializeICU(). 
Aborted (core dumped)
~~~~

I've also found that generally the app doesn't "just run" on Linux. To get it to work I've had to use:
~~~~
chmod u+x "./Boats Animator"
"./Boats Animator"
~~~~

**Comment by Cribstone** [comment]
(Cribstone aka "Humblehacker")-
On v.0.7.0 there seems to be some issues with the naming conventions after the zip package is extracted.
`chmod u+x "./Boats Animator"
"./Boats Animator"`
returns-
`chmod: cannot access ‘./Boats Animator’: No such file or directory
`
Chmod will work if I change the file name to "Boats_Animator" but attempting to execute returns-
` error while loading shared libraries: libnw.so: cannot open shared object file: No such file or directory
`
Perhaps the solution could be as simple as renaming all the packages in the zip file?

**Comment by BoatsAreRockable** [comment]
Thanks for the suggestion! :smile:  Changing the output package name appears to fix the execution errors, although permission changes are still required to make the program execute. Please try out the release I've just tagged to see if it works for you:
https://github.com/BoatsAreRockable/Boats-Animator/releases/tag/0.7.1-alpha1

**Comment by Cribstone** [comment]
Glad to see it should be a simple fix.  Unfortunately, version 0.7.1 unpacks into the exact same structure as previous versions.
  Let me know if anything looks off in my ls output-
```
Boats-Animator-0.7.1-alpha1-linux64\BoatsAnimator
Boats-Animator-0.7.1-alpha1-linux64\icudtl.dat
Boats-Animator-0.7.1-alpha1-linux64\lib\libffmpeg.so
Boats-Animator-0.7.1-alpha1-linux64\lib\libnode.so
Boats-Animator-0.7.1-alpha1-linux64\lib\libnw.so
Boats-Animator-0.7.1-alpha1-linux64\locales\am.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\ar.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\bg.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\bn.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\ca.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\cs.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\da.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\de.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\el.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\en-GB.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\en-US.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\es-419.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\es.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\et.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\fa.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\fil.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\fi.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\fr.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\gu.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\he.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\hi.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\hr.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\hu.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\id.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\it.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\ja.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\kn.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\ko.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\lt.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\lv.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\ml.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\mr.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\ms.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\nb.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\nl.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\pl.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\pt-BR.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\pt-PT.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\ro.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\ru.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\sk.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\sl.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\sr.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\sv.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\sw.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\ta.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\te.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\th.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\tr.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\uk.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\vi.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\zh-CN.pak
Boats-Animator-0.7.1-alpha1-linux64\locales\zh-TW.pak
Boats-Animator-0.7.1-alpha1-linux64\natives_blob.bin
Boats-Animator-0.7.1-alpha1-linux64\nw_100_percent.pak
Boats-Animator-0.7.1-alpha1-linux64\resources.pak
Boats-Animator-0.7.1-alpha1-linux64\snapshot_blob.bin

```

**Comment by BoatsAreRockable** [comment]
@Cribstone it's the same structure, however for some reason it now works

**Comment by Cribstone** [comment]
Can you possibly walk through each step you went through from download to execution?  I'd like to see if I can get it to work using the exact same steps you used.

**Comment by BoatsAreRockable** [comment]
Hi @Cribstone,
Many apologies for the delay, here's how I got it to work:
1. Download the 0.7.1-alpha1 Linux package from [here](https://github.com/BoatsAreRockable/Boats-Animator/releases/tag/v0.7.1-alpha1)
2. Extract the downloaded zip directory.
3. Open a terminal window in the extracted directory and run:
  * `chmod a+x ./BoatsAnimator` to allow the application to execute
  * `./BoatsAnimator` to open the application.



#### Creating a .desktop entry
To save yourself from having to open the terminal every time you wish to use Boats Animator, create a file titled `boats-animator.desktop` in the program's directory containing:
~~~~
[Desktop Entry]
Version=0.7.1
Name=Boats Animator
Exec=BoatsAnimator
Terminal=false
Type=Application
~~~~
And finally set the .desktop file as executable using `chmod a+x boats-animator.desktop`.

I believe the permissions issues are occurring because I am creating the binaries on a Windows machine. I may consider changing this in the future.

I will also be looking to include a .desktop file in future releases as well as a .deb download option.

**Comment by BoatsAreRockable** [comment]
I've just released version 0.7.2 and which fixes the permission issues and adds a .desktop entry. Unzipping the Linux binary and clicking the Boats Animator .desktop entry should execute the application without any further modifications.
